cli/src/config.rs

```diff
@@ -590,10 +610,24 @@ with_env = { env = { KEY1 = "value1", KEY2 = "value2" }, command = ["emacs", "-n
         assert_eq!(name, "emacs");
         assert_eq!(args, ["-nw"].as_ref());
 
+        let command_args: CommandNameAndArgs = config.get("with_env_default").unwrap();
```

Comment:
I was going to suggest adding a test with https://doc.rust-lang.org/std/env/fn.set_var.html here, but I'm not so sure after reading the safety warning there. `cargo-nextest` is, in principle, multithreaded.

You could also set up a test that sets some variable that very likely exists, like PATH or CARGO. I guess to make the test reliable, you could check if that variable is set first, and skip the test if it isn't.